lib/src/validator/flutter_plugin.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:pub_semver/pub_semver.dart';
+
+import '../entrypoint.dart';
+import '../validator.dart';
+
+/// Validates that Flutter plugins doesn't use both new and old plugin format.
+///
```

Comment:
Add a TODO to point to a document describing the formats when it becomes available.